OpenEXR/lib/IlmMacros.cmake

```diff
@@ -0,0 +1,52 @@
+# Means we don't need to set the openexr directory everywhere
+# Uses ${Package}_ROOT
+# Sets ${Package}_INCLUDE_DIR
+macro(_ILM_FIND_INCLUDE_DIR package header)
+	FIND_PATH(${package}_INCLUDE_DIR
+		NAMES ${header}
+		PATH_SUFFIXES include/OpenEXR
+		PATHS ${${package}_ROOT} $ENV{PROGRAMFILES}/openexr
+		)
+	mark_as_advanced(${package}_INCLUDE_DIR)
+endmacro()
+
+# Finds each specified library
+# Uses ${Package}_ROOT, ${Package}_USE_STATIC_LIBS
+# Sets ${Package}_LIBRARIES
+macro(_ILM_FIND_LIBRARIES package components)
+	string(TOLOWER ${package} lowerpackage)
+
+	# Adapted from FindBoost.cmake
+	# Support preference of static libs by looking for static libs first
+	if(${package}_USE_STATIC_LIBS)
+		set(_${lowerpackage}_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
+		if(WIN32)
+			set(CMAKE_FIND_LIBRARY_SUFFIXES .lib .a ${CMAKE_FIND_LIBRARY_SUFFIXES})
+		else()
+			set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
+		endif()
+	endif()
+
+	set(_${lowerpackage}_LIB_VARS)
+	set(${package}_LIBRARIES)
+	foreach(COMPONENT ${_${lowerpackage}_COMPONENTS})
+		string(TOUPPER ${COMPONENT} UPPERCOMPONENT)
+
+		find_library(${package}_${UPPERCOMPONENT}_LIBRARY
+			NAMES ${COMPONENT}
+			PATHS ${${package}_ROOT}/lib $ENV{PROGRAMFILES}/openexr/lib
+			)
```

Comment:
Thanks for taking a stab at this.  I saw your post on the openexr list and this caught my eye since I previously took a stab at tweaking the cmake setup for openexr.  I have a few small review notes if you're interested, but these are purely my opinions and the openexr team may have other preferences.

On RHEL, Fedoras, and possibly other distros (SUSE?) `lib` is not the conventional name for that platform's library directory.  They call it `lib64` for x86_64 machines.  Some multi-arch systems use e.g. `lib/x86_64-linux-gnu`. cmake has a built-in way set of macros that helps deal with these differences[1].  Include:

    include(GNUInstallDirs)

and then use the `${CMAKE_INSTALL_LIBDIR}` variable instead of hard-coding `lib` here.  That should do the right thing on Debian/Ubuntu, Fedora/Redhat, etc.

My other note is really minor but probably helpful for consistency.  I noticed that the variable we're expected to define externally is called `IlmBase_ROOT` (camelcase + allcaps) because it's registered as `project(IlmBase)` in CMakeListst.txt.

Would it make sense to register it as `project(ILMBASE)` so that we can use more traditional and consistent all-caps variable names for all of the external variables?

It seems helpful since it'd make it consistent with e.g. `ZLIB_LIBRARIES`, etc., so it should be more intuitive for existing cmake users.

I'd personally like to see the cmake setup expanded for openexr.  Another thing that is not completely exposed through the cmake setup is the custom namespacing support  (see README.namespacing and the corresponding configure.ac settings).

It'd be really nice to get the cmake setup to the point where it can fully replace the autoconf/configure system.  I'm not sure if that's desired, but it seems like the right thing to do from a maintenance POV.

cheers